Source/CBLSyncConnection.m

```diff
@@ -265,6 +266,22 @@ - (void) removeAttachmentProgress: (NSProgress*)attProgress
 }
 
 
+- (BOOL) accessCheckForRequest: (BLIPRequest*)request
+                         docID: (NSString*)docID
+{
+    if (self.onSyncAccessCheck) {
+        CBLStatus status = self.onSyncAccessCheck(request, docID);
+        if (CBLStatusIsError(status)) {
+            NSString* message;
+            [request respondWithErrorCode: CBLStatusToHTTPStatus(status, &message)
```

Comment:
This is a race condition: it's not guaranteed that CBLStatusToHTTPStatus will have been called before the `message` parameter is pushed on the stack. You'll need to break the function call out and put it above this line.